cartographer/mapping_3d/imu_integration.h

```diff
@@ -36,25 +36,19 @@ struct IntegrateImuResult {
   Eigen::Quaternion<T> delta_rotation;
 };
 
-// Returns velocity delta in map frame.
-IntegrateImuResult<double> IntegrateImu(
-    const std::deque<sensor::ImuData>& imu_data, const common::Time start_time,
-    const common::Time end_time,
-    std::deque<sensor::ImuData>::const_iterator* it);
-
-template <typename T>
+template <typename T, typename RangeType, typename IteratorType>
 IntegrateImuResult<T> IntegrateImu(
-    const std::deque<sensor::ImuData>& imu_data,
+    const RangeType& imu_data,
     const Eigen::Transform<T, 3, Eigen::Affine>&
         linear_acceleration_calibration,
     const Eigen::Transform<T, 3, Eigen::Affine>& angular_velocity_calibration,
     const common::Time start_time, const common::Time end_time,
-    std::deque<sensor::ImuData>::const_iterator* it) {
+    IteratorType* const it) {
```

Comment:
Have you considered the alternative "ContainerType::const_iterator*? This would require less abstraction.

Reply:
This change is to support ranges, e.g. `mapping::Range<>`, so `const_iterator` is not available.